IntervalTree.h

```diff
@@ -21,6 +21,24 @@ class Interval {
     , stop(std::max(s, e))
     , value(v) 
     {}
+
+    Interval()
+    {
+    }
+
+    constexpr bool operator==(const Interval& other) const noexcept
+    {
+        return start == other.start &&
+            stop == other.stop &&
+            value == other.value;
+    }
+
+    constexpr bool operator!=(const Interval& other) const noexcept
+    {
+        return start != other.start ||
+            stop != other.stop ||
+            value != other.value;
```

Comment:
```suggestion
        return !(*this == other);
```

Leveraging the equality operator here helps reduce redundancy :)